src/client/dfs/dfs.c

```diff
@@ -1507,6 +1518,162 @@ dfs_cont_create_with_label(daos_handle_t poh, const char *label, dfs_attr_t *att
 	return rc;
 }
 
+int
+dfs_connect(const char *pool, const char *sys, const char *cont, int flags, dfs_attr_t *attr,
+	    dfs_t **_dfs)
+{
+	daos_handle_t		poh = {0};
+	daos_handle_t		coh = {0};
+	bool			pool_h_bump = false;
+	bool			cont_h_bump = false;
+	struct dfs_mnt_hdls	*pool_hdl = NULL;
+	struct dfs_mnt_hdls	*cont_hdl = NULL;
+	dfs_t			*dfs = NULL;
+	int			amode, cmode;
+	int			rc, rc2;
+
+	if (_dfs == NULL)
+		return EINVAL;
+
+	if (!dfs_is_init()) {
+		D_ERROR("dfs_init() must be called before dfs_connect() can be used\n");
+		return EACCES;
+	}
+
+	amode = (flags & O_ACCMODE);
+
+	pool_hdl = dfs_hdl_lookup(pool, DFS_H_POOL);
```

Comment:
Do we need to also enforce pool is not NULL, since it will be passed to the hashing functions? (I'm not sure if it's an issue)

Reply:
yes indeed will add a check.